Yoast/wordpress-seo
# Prevent errors on frontend

## Context


* When errors occured during the migrations errors might appear on the front end

## Summary


This PR can be summarized in the following changelog entry:

* Fixes a bug where errors are shown on frontend page when the tables are not present

## Relevant technical choices:

*

## Test instructions

This PR can be tested by following these steps:

1. Make sure  `YOAST_ENVIRONMENT` is set to production otherwise the exception still will be thrown. `define( 'YOAST_ENVIRONMENT', 'production' );`
2. On the release branch (14.1)
3. Remove all indexable related tables and the option (`yoast_migrations_free`) in the options table.
4. Now create a table named: `wp_yoast_indexable`
```sql
CREATE TABLE `wp_yoast_indexable` (
  `id` int(11) unsigned NOT NULL AUTO_INCREMENT,
  PRIMARY KEY (`id`)
) ENGINE=InnoDB DEFAULT CHARSET=latin1;
```
5. Visit the Yoast SEO dashboard / WordPress admin dashboard or any other Yoast page in the admin. This will trigger the migrations.
6. Visit the home page of your test site and see all kind of errors occur.
7. If you are able navigate to some post
8. Checkout this branch
9. Repeat step 3.
10. See the errors aren't visible on the page itself (maybe query monitor is collecting them)
11. Make sure the breadcrumbs and meta data is calculated right.

## UI changes

## Documentation

## Quality assurance


Fixes #15044
